pennylane/workflow/qnode.py

```diff
@@ -735,18 +758,16 @@ def best_method_str(device, interface):
         return transform
 
     @staticmethod
-    def _validate_backprop_method(device, interface, shots=None):
-        if shots is not None or _get_device_shots(device):
-            raise qml.QuantumFunctionError("Backpropagation is only supported when shots=None.")
-
+    def _validate_backprop_method(device, interface, tape=None):
         if isinstance(device, qml.devices.Device):
-            config = qml.devices.ExecutionConfig(gradient_method="backprop", interface=interface)
-            if device.supports_derivatives(config):
-                return "backprop", {}, device
-            raise qml.QuantumFunctionError(
-                f"Device {device.name} does not support backprop with {config}"
+            raise ValueError(
+                "QNode._validate_backprop_method only applies to the qml.Device interface."
             )
+        if tape.shots if tape else _get_device_shots(device):
+            raise qml.QuantumFunctionError("Backpropagation is only supported when shots=None.")
 
+        if tape and any(isinstance(m.obs, qml.SparseHamiltonian) for m in tape.measurements):

```

Comment:
Moved from `QNode.construct` and `QNode.gradient_fn` is not accurate at that point in the call.